pkg/tfgen/docs.go

```diff
@@ -198,10 +200,36 @@ func getDocsForProvider(g *generator, org string, provider string, resourcePrefi
 	return doc, nil
 }
 
+// checkIfNewDocsExist checks if the new docs root exists
+func checkIfNewDocsExist(repo string) bool {
+	// Check if the new docs path exists
+	newDocsRoot := path.Join(repo, "docs")
+	_, err := os.Stat(newDocsRoot)
+	return !os.IsNotExist(err)
+}
+
+// getDocsPath finds the correct docs path for the repo/kind
+func getDocsPath(repo string, kind DocKind) string {
+	// Check if the new docs path exists
+	newDocsExist := checkIfNewDocsExist(repo)
+
+	if !newDocsExist {
+		// If the new path doesn't exist, use the old docs path.
+		kindString := string([]rune(kind)[0]) // We only want the first letter because the old path uses "r" and "d"
+		return path.Join(repo, "website", "docs", kindString)
+	}
+
+	// Otherwise use the new location path.
+	kindString := string(kind)
+	return path.Join(repo, "docs", kindString)
+}
+
 // readMarkdown searches all possible locations for the markdown content
 func readMarkdown(repo string, kind DocKind, possibleLocations []string) ([]byte, string, bool) {
+	locationPrefix := getDocsPath(repo, kind)
+
 	for _, name := range possibleLocations {
-		location := path.Join(repo, "website", "docs", string(kind), name)
+		location := path.Join(locationPrefix, name)
```

Comment:
Nit: I realize this is an existing use of `path.Join` and we run this code always on *nix so it doesn't really matter, but it'd be more correct to use `filepath.Join` when joining file paths.